lib/api/src/entities/engine/inner.rs

```diff
@@ -0,0 +1,156 @@
+                "The selected runtime does not support `deserialize_unchecked`".into(),
+            )),
+        }
+    }
+
+    #[cfg(all(feature = "sys", not(target_arch = "wasm32")))]
+    /// Deserializes a WebAssembly module which was previously serialized with
+    /// `Module::serialize`,
+    ///
+    /// # Errors
+    /// Not every implementer supports serializing and deserializing modules.
+    /// Currently, only the `sys` engines support it, and only when the target
+    /// architecture is not `wasm32`.
+    pub(crate) unsafe fn deserialize(
+        &self,
+        bytes: impl IntoBytes,
+    ) -> Result<Arc<Artifact>, DeserializeError> {
+        match self {
+            #[cfg(feature = "sys")]
+            Self::Sys(s) => s.deserialize(bytes.into_bytes().to_owned().into()),
+            _ => Err(DeserializeError::Generic(
+                "The selected runtime does not support `deserialize`".into(),
+            )),
+        }
+    }
+
+    #[cfg(all(feature = "sys", not(target_arch = "wasm32")))]
+    /// Load a serialized WebAssembly module from a file and deserialize it.
+    ///
+    /// # Note
+    /// You should almost always prefer [`Self::deserialize_from_file`].
+    ///
+    /// # Errors
+    /// Not every implementer supports serializing and deserializing modules.
+    /// Currently, only the `sys` engines support it, and only when the target
+    /// architecture is not `wasm32`.
+    ///
+    /// # Safety
+    /// See [`Artifact::deserialize_unchecked`].
+    pub(crate) unsafe fn deserialize_from_file_unchecked(
+        &self,
+        file_ref: &Path,
+    ) -> Result<Arc<Artifact>, DeserializeError> {
+        match self {
+            #[cfg(feature = "sys")]
+            Self::Sys(s) => s.deserialize_from_file_unchecked(file_ref),
+            _ => Err(DeserializeError::Generic(
+                "The selected runtime does not support `deserialize_from_file_unchecked`".into(),
+            )),
+        }
+    }
+
+    #[cfg(all(feature = "sys", not(target_arch = "wasm32")))]
+    /// Load a serialized WebAssembly module from a file and deserialize it.
+    ///
+    /// # Errors
+    /// Not every implementer supports serializing and deserializing modules.
+    /// Currently, only the `sys` engines support it, and only when the target
+    /// architecture is not `wasm32`.
+    ///
+    /// # Safety
+    /// See [`Artifact::deserialize`].
+    pub(crate) unsafe fn deserialize_from_file(
+        &self,
+        file_ref: &Path,
+    ) -> Result<Arc<Artifact>, DeserializeError> {
+        match self {
+            #[cfg(feature = "sys")]
+            Self::Sys(s) => s.deserialize_from_file(file_ref),
+            _ => Err(DeserializeError::Generic(
+                "The selected runtime does not support `deserialize_from_file`".into(),
+            )),
+        }
+    }
+}
+
+impl Default for RuntimeEngine {
+    #[allow(unreachable_code)]
+    fn default() -> Self {
+        #[cfg(feature = "sys")]
```

Comment:
One thing is enabling a `sys` runtime, and other a default runtime being `sys`. I think we need a new feature that let you choose the default (eg: `sys-default`), that can't be activated when other default runtime (eg: `js-default`) or similar are activated (compiler error)

Reply:
We already have a feature called `sys-default`. What would happen if `sys` is enabled but not `sys-default`?

Reply:
Basically, we should not have default for `RuntimeEngine` if there's no default defined

Reply:
If we don't want to have that because is a breaking change, then I'd have what you have now as a fallback, not as the main path

Reply:
Yup

Reply:
[c111880](https://github.com/wasmerio/wasmer/pull/5243/commits/c11188021375a0e74cfb1fe9754c7524f40f0fbb)